Bio/Seq.py

```diff
@@ -1319,23 +1318,35 @@ def ungap(self, gap=None):
     def join(self, other):
         """Return a merge of the sequences in other, spaced by the sequence from self.
 
-        Accepts all Seq objects and Strings as objects to be concatenated with the spacer
+        Accepts either a Seq or string (and iterates over the letters), or an
+        iterable containing Seq or string objects. These arguments will be
+        concatenated with the calling sequence as the spacer:
 
         >>> concatenated = Seq('NNNNN').join([Seq("AAA"), Seq("TTT"), Seq("PPP")])
         >>> concatenated
         Seq('AAANNNNNTTTNNNNNPPP')
 
-        Throws error if other is not an iterable and if objects inside of the iterable
-        are not Seq or String objects
+        Joining the letters of a single sequence:
+
+        >>> Seq('NNNNN').join(Seq("ACGT"))
+        Seq('ANNNNNCNNNNNGNNNNNT')
+        >>> Seq('NNNNN').join("ACGT")
+        Seq('ANNNNNCNNNNNGNNNNNT')
         """
-        if not isinstance(other, collections.abc.Iterable):
-            # doesn't detect single strings
-            raise ValueError("Input must be an iterable")
-        if isinstance(other, str):
-            raise ValueError("Input must be an iterable")
         from Bio.SeqRecord import SeqRecord  # Lazy to avoid circular imports
 
         a = self.alphabet
+        if isinstance(other, (Seq, MutableSeq, UnknownSeq)):
```

Comment:
Technically `UnknownSeq` not needed here (since it subclasses `Seq`) but perhaps better to be explicit anyway

Reply:
Fair point.

Reply:
Fixing that now...